sources/PPU/PPU.cpp

```diff
@@ -535,62 +543,66 @@ namespace ComSquare::PPU
 
 	Vector2<bool> PPU::getBackgroundMirroring(int bgNumber) const
 	{
-		Vector2<bool> backgroundSize(false, false);
-
-		backgroundSize.y = this->_registers._bgsc[bgNumber - 1].tilemapVerticalMirroring;
-		backgroundSize.x = this->_registers._bgsc[bgNumber - 1].tilemapHorizontalMirroring;
-		return backgroundSize;
+		return {
+			static_cast<bool>(this->_registers._bgsc[bgNumber - 1].tilemapHorizontalMirroring),
+			static_cast<bool>(this->_registers._bgsc[bgNumber - 1].tilemapVerticalMirroring)
+		};
 	}
 
 	void PPU::renderMainAndSubScreen()
 	{
 		uint16_t colorPalette;
 		// should only render backgrounds needed (depending of th bgMode)
-		int i = 0;
 		for (auto &_background : this->_backgrounds) {
-			i++;
 			_background.renderBackground();
 		}
 		// TODO make a function getDefaultBgColor
 		colorPalette = this->cgram.read(0);
 		colorPalette += this->cgram.read(1) << 8U;
 
-		uint32_t color = Utils::getRealColor(colorPalette);
+		uint32_t color = Utils::CGRAMColorToRGBA(colorPalette);
 		for (auto &row : this->_subScreen)
 			row.fill(color);
+		for (auto &row : this->_mainScreenLevelMap)
+			row.fill(0);
+		for (auto &row : this->_subScreenLevelMap)
+			row.fill(0);
 		// the buffer is overwrite if necessary by a new bg so the background priority is from back to front
 		// the starting palette index isn't implemented
 		switch (this->_registers._bgmode.bgMode) {
 		case 0:
-			this->addToMainSubScreen(this->_backgrounds[BgName::bg4NoPriority]);
-			this->addToMainSubScreen(this->_backgrounds[BgName::bg3NoPriority]);
+			this->addToMainSubScreen<0, 15>(this->_backgrounds[BgName::Background4]);
```

Comment:
I have no idea why, but I guess this is good.

Reply:
0 and 15 are levelValues for low and high priority respectively
it's like the z index in css for an element
the element with the higher z index will be displayed on top